tests/fast/FuzzApiCorrectness.toml

```diff
@@ -1,7 +1,7 @@
 [configuration]
 StderrSeverity = 30
-allowDisablingTenants = false
 allowDefaultTenant = false
+tenantModes = [1]
```

Comment:
I suppose for these tests we should probably allow it to choose between optional and required, we just don't want it to be disabled.

Reply:
That would be different behaviour than what was specified previously though. If provided
`allowDefaultTenant = false`
`allowDisablingTenants = false`
before my changes, it was guaranteed to be `optional` every time

Reply:
Yeah, you're right. I think for some tests, we may want to actually expand this now that we have the option to (e.g. all the tenant management and metacluster management tests should work in both optional and required, I think), but others may not work in required mode.

Reply:
So change all the `Tenant*` workloads to pick between optional and required, but leave it at only optional for the rest?

Reply:
The higher level guiding principle would be that we should test all of the modes that the test supports, and for a test that is creating and using tenants the `required` mode would be legal. I can't comment with certainty on the other tests, but if a quick look reveals them to also be strictly tenant based then we could do the same for them.